### DISCUSSION TITLE
How can access to ContainedPart items?

### DISCUSSION THREAD

**Discussion by overstartup** [OrchardCMS/OrchardCore#9312] [Q&A]
i have  "story" content, and it has a list of "slides". 
How can access to the list?
```
Last = {"ContainedPart": {
  "ListContentItemId": "4nggvcq5mrmgy4s8baywysc1cj",
  "Order": 1
}}
```

im using this code to get stories:
```
 var queryProducts = _session.Query<ContentItem, ContentItemIndex>();
                queryProducts = queryProducts.With<ContentItemIndex>(x => x.Latest);
                queryProducts = queryProducts.With<ContentItemIndex>(x => x.Published);
                queryProducts = queryProducts.With<ContentItemIndex>(x => x.ContentType == "Story");

                queryProducts = queryProducts.OrderBy(cr => cr.DisplayText);
                var contentItemProducts = await queryProducts.ListAsync();

                var contentItemSummaries = new List<StoryItem>();
                foreach (var contentItem in contentItemProducts)
                {
                    StoryItem Product = new StoryItem();
                    Product = await convertStoryAsync(contentItem);
                    contentItemSummaries.Add(Product);
                }
```

i can not find 'ListContentItemId' in contentItem to retrieve list of slides.

**Comment by jtkech** [comment]
There is the `ContainedPartIndex` which allows to retrieve all items (here the slides) contained in a given list (here a story) whose `ContentItemId` is equal to the index `ListContentItemId` column.

**Reply by overstartup** [reply]
thank you for help.
it solved by:
```
 var queryslide = _session.Query<ContentItem, ContainedPartIndex>();
            queryslide = queryslide.With<ContainedPartIndex>(x => x.ListContentItemId == reg.ContentItemId);
            queryslide = queryslide.OrderBy(cr => cr.Order); 
           var contentItemProducts = await queryslide.ListAsync();
```

just i have a couple question:

1-how can i add condition for latest and published?
2- i have "agegroup" filed . How I can set a condition for this field?

**Reply by jtkech** [reply]
For 1. You need to add another `.With` on the `ContentItemIndex` with the additional predicate.

For 2. You would need another index table including this field, there is the SQL fields feature that provides index tables for all default fields but it adds a lot of index tables. Or if there is not so much items for a given story, just does the query and then filter on the query results.
